### ISSUE TITLE
Mouse lasso selection list view

### ISSUE THREAD

**Issue by orschiro** [linuxmint/nemo#416]
With nemo 2.0 it is not yet possible to select multiple files with a mouse lasso in list view mode. If I press and hold the left mouse button and try to drag a lasso around the files, nothing actually happens. Is the implementation of that feature something you foresee for the future?

![lasso-selection-nemo](https://f.cloud.github.com/assets/170947/1320694/b5d8a08e-3370-11e3-88fc-cfb81e333e9a.png)

**Comment by jensb** [comment]
Where exactly is the complexity? The functionality is already in icon and compact view. 
(This is not rethoric, I really want to understand. Perhaps you can describe what would have to be done, so that a junior coder can attempt to fix it.)

**Comment by anandrkris** [comment]
@jensb - Am not a developer, just a noob comment made without reading the thread fully. I prolly didn't notice that it already implemented in other views when I posted.

**Comment by Jo-X** [comment]
+1 
this would be a really handy feature and I am missing it since switching from pcmanfm (lxde)

**Comment by Arnopello** [comment]
FWIW... +1
I even thought I broke something in Nemo :p

**Comment by Kosyne** [comment]
+1 as well. Doing the ole' shift+click, ctrl+shift+click, etc, is not fun when you have only a touchpad available. Only thing forcing me to use other views as my primary.

**Comment by schlegel11** [comment]
+1

**Comment by jf99** [comment]
+1

**Comment by andrew160** [comment]
+1

**Comment by kmkom** [comment]
+1 (Please)

**Comment by VinnyFonseca** [comment]
+1

**Comment by guzzard** [comment]
+1

**Comment by jensb** [comment]
Can one of the developers please comment on what is holding this issue back? There are two solutions "out there", both work fine and one of them could be duplicated: 

- either make only the "Name" column selectable and treat all other columns as "free space" when dragging the mouse to create a select rectangle (Windows)
- Or just select every file the mouse is being dragged over without displaying a rectangle at all (Mac)

Both solutions make sense for touch enabled usage too.

**Comment by shvfn** [comment]
+1

**Comment by redspyder** [comment]
+1 Windows  > Linux in this regard.

**Comment by Fopedush** [comment]
+1

absurd that this feature is missing

**Comment by jensb** [comment]
Still not there. What is the issue? Please, at least tell us what is holding this issue back. Can the feature be  ported from Thunar to Nemo?

**Comment by JosephMcc** [comment]
> What is the issue? Please, at least tell us what is holding this issue back.

Yeah, no one has decided to pick it up and work on it. I have a feeling it's probably not simple to get working and, more importantly, working correctly and without issues. In addition, the end result this would provide is already easily accomplished by use of Ctrl+Click. That means a lot of work vs. small gain, so not a huge priority.

**Comment by jensb** [comment]
MAybe, but "small gain" * 100.000 users is still a lot. Ctrl-Click for 20 files means 20 times the effort plus having to reach the keyboard. All other file managers, especially (!) the commercial ones, can do it, so it's an expected feature, not something that people will ignore if it isn't there.
Maybe somebody will dive in if there's a rough explanation of what would have to be done.

**Comment by JosephMcc** [comment]
> Ctrl-Click for 20 files means 20 times the effort plus having to reach the keyboard.

Do you even know how Ctrl+Click works in the list view? Click where you want the selection to start and Ctrl+Click where you want it to end. That's hardly 20 clicks.

**Comment by jensb** [comment]
No(t here). You probably meant Shift+Click. I can also understand that developers don't need this feature often since they do most of their work on the keyboard anyway, but that doesn't apply to non-developers. Just look at the number of votes for this issue.
So what would have to be done? Maybe I'll dive into the code myself.

**Comment by JosephMcc** [comment]
Sorry you're right. I did mean shift click.

**Comment by cschreib** [comment]
Related to this. Reducing the "selectable" area to the name of the file only (rather than the entire row) would also make it easier to reach the right-click context menu for the current folder (this menu is really useful, for example to "Create new folder" or "Open in terminal" or what have you).

Right now, this almost works: if you right click on a file's name (note: the name, not the row) it brings this file's context menu. If you right click outside of the name (in the blank space of the "name" column, or in any other column) then the context menu of the current folder pops up... except if you did the right-click on the row of the currently selected file! For this file, and this file only, right-clicking anywhere on the entire row will bring the file's context menu, and never that of the current folder. This is quite confusing, and it took me some time to find that I had to right click on another file's row for it to work.

**Comment by qoaa** [comment]
this would be awesome... i was trying to do this earlier with a huge collection of music to organize each album into subdirectories by album name. 

i would love to see lasso selection with mouse. I'm use to it in every other filemanager ive seen personally but i love mint too much hehe

**Comment by jonpovey** [comment]
+1 another suprised user coming from Windows, disappointed that this feature doesn't exist. Yes I can use shift and ctrl and sometimes do (in Windows as well) but it feels clunky and awkward that lasso doesn't work here.

**Comment by JosephMcc** [comment]
Many thanks for contributing to Nemo. Your suggestion was reviewed.

For more information on our workflow and feature requests, read https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/faq.html.

**Comment by jensb** [comment]
So does this mean it will be implemented?

**Comment by guzzard** [comment]
@jensb, it says right here what you can expect.. https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/faq.html.

**Comment by jensb** [comment]
It says nothing about further tracking of this issue, however. (I just edited my comment..)

**Comment by schlegel11** [comment]
This is a good question. What does it mean to close the issue?
For me this method is a little bit non transparent.
So can we see the future roadmap or something like that ?

**Comment by JosephMcc** [comment]
We are making an effort to clean up our issue trackers. Closed doesn't really mean it will be implemented or not but there is no reason for these to sit here opened for years. We are well aware that this is a requested feature but I wouldn't get my hopes since it's been here for nearly 5 years without being done.

**Comment by schlegel11** [comment]
Ah ok this sounds a little clearer. For me it might be a good idea or a more transparent one if the issue is labeled with something like "No implementation expected" due to some reasons or so.

**Comment by VinnyFonseca** [comment]
This is shameful. No feedback, no acknowledgement of the issue, no further discussion, no reasons for not implementing, and now a link to the FAQ saying "Thank you for reporting, your job is done, it doesn't mean much and you cannot track what issues have already been submitted".

For the record, I'm fine with not implementing the feature and I absolutely understand not all features are implemented. All I ask for is a transparent update, when what we got was "use ctrl + click" as if it solved the problem.

Various other feature requests I was subscribed to were closed in the same manner. I'll move to another file explorer. I won't contribute to a team that does not respect its users and the best practices of the open source community. https://trello.com/b/lsBmkzPY/ubuntu-desktop-1804-cycle to see how it's done.
